GPflow/ekernels.py

```diff
@@ -0,0 +1,307 @@
+from functools import reduce
+import warnings
+import tensorflow as tf
+from . import kernels
+from .tf_wraps import eye
+from ._settings import settings
+
+int_type = settings.dtypes.int_type
+float_type = settings.dtypes.float_type
+
+# TODO: Efficiency gain from using diagonal q(X).
```

Comment:
I think we should just have a github issue. TODOs in the code make it seem incomplete, which I don't think fairly reflects the status of the code.